asm/unk_0206A360_asm.s

```diff
@@ -11,543 +11,9 @@
-	str r0, [r4, #0x10]
-	mov r3, #0
-	str r3, [sp]
-	mov r0, #0x1c
-	str r0, [sp, #4]
-	mov r2, #2
-	str r2, [sp, #8]
-	mov r0, #0xd
-	str r0, [sp, #0xc]
-	mov r0, #0xb
-	lsl r0, r0, #6
-	str r0, [sp, #0x10]
-	ldr r0, [r4, #0xc]
-	ldr r1, [r4, #0x10]
-	bl AddWindowParameterized
-	add sp, #0x14
-	pop {r3, r4, pc}
-	.balign 4, 0
-	thumb_func_end sub_0206A660
-
-	thumb_func_start sub_0206A694
-sub_0206A694: ; 0x0206A694
-	push {r3, r4, r5, r6, lr}
-	sub sp, #0xc
-	add r5, r1, #0
-	ldr r1, [r0, #0x30]
-	cmp r1, #0
-	bne _0206A6A6
-	add sp, #0xc
-	mov r0, #1
-	pop {r3, r4, r5, r6, pc}
-_0206A6A6:
-	add r4, r0, #0
-	ldrh r0, [r0, #0x34]
-	add r4, #0x34
-	cmp r0, #5
-	bhi _0206A790
-	add r0, r0, r0
-	add r0, pc
-	ldrh r0, [r0, #6]
-	lsl r0, r0, #0x10
-	asr r0, r0, #0x10
-	add pc, r0
-_0206A6BC: ; jump table
-	.short _0206A6C8 - _0206A6BC - 2 ; case 0
-	.short _0206A6D8 - _0206A6BC - 2 ; case 1
-	.short _0206A714 - _0206A6BC - 2 ; case 2
-	.short _0206A72A - _0206A6BC - 2 ; case 3
-	.short _0206A74A - _0206A6BC - 2 ; case 4
-	.short _0206A78A - _0206A6BC - 2 ; case 5
-_0206A6C8:
-	ldr r0, [r5, #0x40]
-	mov r1, #0
-	bl PlayerAvatar_ToggleAutomaticHeightUpdating_NowApply
-	ldrh r0, [r4]
-	add r0, r0, #1
-	strh r0, [r4]
-	b _0206A790
-_0206A6D8:
-	ldr r0, [r5, #0x40]
-	bl PlayerAvatar_GetMapObject
-	ldrh r1, [r4, #2]
-	add r6, r0, #0
-	add r1, r1, #1
-	strh r1, [r4, #2]
-	add r1, sp, #0
-	bl MapObject_GetPositionVec
-	mov r0, #2
-	ldr r1, [sp, #4]
-	lsl r0, r0, #0xc
-	sub r0, r1, r0
-	str r0, [sp, #4]
-	add r0, r6, #0
-	add r1, sp, #0
-	bl MapObject_SetPositionVec
-	ldrh r0, [r4, #2]
-	cmp r0, #0x10
-	blo _0206A790
-	ldr r0, [r5, #0x40]
-	mov r1, #1
-	bl PlayerAvatar_ToggleAutomaticHeightUpdating_NowApply
-	ldrh r0, [r4]
-	add r0, r0, #1
-	strh r0, [r4]
-	b _0206A790
-_0206A714:
-	ldr r0, [r5, #0x40]
-	mov r1, #1
-	bl Field_PlayerAvatar_OrrTransitionFlags
-	ldr r0, [r5, #0x40]
-	bl Field_PlayerAvatar_ApplyTransitionFlags
-	ldrh r0, [r4]
-	add r0, r0, #1
-	strh r0, [r4]
-	b _0206A790
-_0206A72A:
-	ldr r0, [r5, #0x40]
-	bl PlayerAvatar_GetMapObject
-	add r5, r0, #0
-	bl MapObject_AreBitsSetForMovementScriptInit
-	cmp r0, #0
-	beq _0206A790
-	add r0, r5, #0
-	mov r1, #1
-	bl MapObject_SetHeldMovement
-	ldrh r0, [r4]
-	add r0, r0, #1
-	strh r0, [r4]
-	b _0206A790
-_0206A74A:
-	ldr r0, [r5, #0x40]
-	bl PlayerAvatar_GetMapObject
-	bl MapObject_AreBitsSetForMovementScriptInit
-	cmp r0, #0
-	beq _0206A790
-	add r0, r5, #0
-	bl FollowMon_IsActive
-	cmp r0, #0
-	beq _0206A782
-	add r0, r5, #0
-	mov r1, #1
-	bl ov01_02205790
-	add r0, r5, #0
-	bl FollowMon_GetMapObject
-	mov r1, #0x30
-	bl sub_0205FC94
-	add r0, r5, #0
-	bl FollowMon_GetMapObject
-	mov r1, #1
-	bl sub_02069DC8
-_0206A782:
-	ldrh r0, [r4]
-	add r0, r0, #1
-	strh r0, [r4]
-	b _0206A790
-_0206A78A:
-	add sp, #0xc
-	mov r0, #1
-	pop {r3, r4, r5, r6, pc}
-_0206A790:
-	mov r0, #0
-	add sp, #0xc
-	pop {r3, r4, r5, r6, pc}
-	.balign 4, 0
-	thumb_func_end sub_0206A694
-
 	.data
 
+.public _0210FAE8
 _0210FAE8:
```

Comment:
Port to C. Since it's in `.data`, it's non-const, but still static

Reply:
was saving for post dumping the narc, just trying to get it rebuilt at the moment but I got sick lol